sktime/contrib/rotation_forest/rotation_forest_dev.py

```diff
@@ -0,0 +1,202 @@
+from sklearn.ensemble.forest import ForestClassifier
+from sklearn.tree import DecisionTreeClassifier
+from sklearn.ensemble.forest import _generate_sample_indices, compute_sample_weight
+import numpy as np
+from numpy import random
+from sklearn.decomposition import PCA
+from scipy.stats import mode
+from copy import deepcopy
+
+#they extract random subsections of the data, and perform PCA with variancecoverage of 1.0
+#the number of subsections of the data is based on the method,
+#generateGroupsFromNumbers or generateGroupsFromSizes.
+
+class RotationForest(ForestClassifier):
+
+    def __init__(self,
+                 n_estimators=10,
+                 minGroup=3,
+                 maxGroup=3,
+                 remove_percentage=50,
+                 random_state=None,
+                 verbose=0,):
+        super(RotationForest, self).__init__(
+            base_estimator=DecisionTreeClassifier(),
+            n_estimators=n_estimators)
+
+        self.verbose = verbose
+        self.random_state = random_state
+        self.minGroup = minGroup
+        self.maxGroup = maxGroup
+        self.remove_percentage = remove_percentage
+        self.base_projectionFilter = PCA()
+        self.base_estimator = DecisionTreeClassifier()
+        self._std = []
+        self._mean = []
+        self._noise = []
+        self._classifiers = []
+        self._num_classes = 0
+        self._num_atts = 0
+        self.classes_ = None
+
+        self._groups = [None] * self.n_estimators
+        self._pcas = [None] * self.n_estimators
+
+        random.seed(random_state)
```

Comment:
Would use `rng = check_random_state(random_state)` from sklearn instead and call all functions that depend on it directly on the `rng` object, e.g. `rng.uniform()`